pkg/db/v1beta1/mysql/mysql.go

```diff
@@ -31,19 +30,73 @@ type dbConn struct {
 	db *sql.DB
 }
 
-func getDbName() string {
-	dbPassEnvName := common.DBPasswordEnvName
-	dbPass := os.Getenv(dbPassEnvName)
+func CreateMySQLConfig(user, password string, mysqlServiceHost string,
+	mysqlServicePort string, dbName string, mysqlExtraParams map[string]string) *mysql.Config {
+
+	params := map[string]string{}
+
+	for k, v := range mysqlExtraParams {
+		params[k] = v
+	}
+
+	return &mysql.Config{
+		User:                 user,
+		Passwd:               password,
+		Net:                  "tcp",
+		Addr:                 fmt.Sprintf("%s:%s", mysqlServiceHost, mysqlServicePort),
+		Params:               params,
+		DBName:               dbName,
+		AllowNativePasswords: true,
+		Timeout:              connectInterval,
+	}
+}
+
+func initDB() (string, error) {
+	dbPass := os.Getenv(common.DBPasswordEnvName)
 	dbUser := env.GetEnvOrDefault(
 		common.DBUserEnvName, common.DefaultMySQLUser)
 	dbHost := env.GetEnvOrDefault(
-		common.MySQLDBHostEnvName, common.DefaultMySQLHost)
+		common.DBHostEnvName, common.DefaultMySQLHost)
 	dbPort := env.GetEnvOrDefault(
-		common.MySQLDBPortEnvName, common.DefaultMySQLPort)
-	dbName := env.GetEnvOrDefault(common.MySQLDatabase,
+		common.DBPortEnvName, common.DefaultMySQLPort)
+	dbName := env.GetEnvOrDefault(common.DBNameEnvName,
 		common.DefaultMySQLDatabase)
+	dbExtraParamsStr := env.GetEnvOrDefault(common.DBExtraParamsEnvName,
+		common.DefaultMySQLExtraParams)
+
+	dbExtraParams := make(map[string]string)
+	err := json.Unmarshal([]byte(dbExtraParamsStr), &dbExtraParams)
+	if err != nil {
+		klog.Errorf("Failed to get dbExtraParams. Error: %v", err)
+	}
 
-	return fmt.Sprintf(dbNameTmpl, dbUser, dbPass, dbHost, dbPort, dbName)
+	if dbPass != "" {
+		mysqlConfig := CreateMySQLConfig(dbUser, dbPass, dbHost, dbPort, "", dbExtraParams)
+
+		ticker := time.NewTicker(connectInterval)
+		defer ticker.Stop()
+
+		timeoutC := time.After(connectTimeout)
+		for {
+			select {
+			case <-ticker.C:
+				if db, err := sql.Open(dbDriver, mysqlConfig.FormatDSN()); err == nil {
+					if _, err = db.Exec(fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s", dbName)); err == nil {
+						mysqlConfig.DBName = dbName
```

Comment:
Do you need to set `DBName` here since you are doing this in the `CreateMySQLConfig`: https://github.com/kubeflow/katib/blob/3e2d6fc72eecbb0668e2be8c4bbbc83ff1ba61fd/pkg/db/v1beta1/mysql/mysql.go#L48 ?